qbraid_qir/qasm3/convert.py

```diff
@@ -69,14 +81,23 @@ def qasm3_to_qir(
 
     final_module = QasmQIRModule(name, qasm3_module, llvm_module)
 
-    profile = profile.lower()
+    # Convert enum to string if needed, then normalize and validate
+    if isinstance(profile, Profile):
+        profile_str = profile.value.lower()
+    else:
+        profile_str = profile.lower()
+        # Validate that the string corresponds to a valid profile
+        valid_profiles = [p.value for p in Profile]
```

Comment:
Let's allow the user to specify either `str` or `Profile` but internal to the repo let's use the `Enum`. Also, we can move the validation logic inside the `Profile` class itself rather than placing it here